src/expertsystem/reaction/__init__.py

```diff
@@ -596,14 +604,17 @@ def check(
         topology_building=topology_building,
     )
 
-    if any(map(len, results.violated_rules.values())):
+    if any(map(len, results.violated_node_rules.values())) or any(
```

Comment:
Hmm maybe
```python
violated_rules = (
    results.violated_node_rules.values()
    + results.violated_edge_rules.values()
)
if any(map(len, violated_rules)):
    ...
```
Ah but there we go again, this `Result` class is just fishy...

Reply:
I tried that as well, but you cant add generators, so I would have to wrap each part into a `list()`.